src/kOS/Utilities/VesselUtils.cs

```diff
@@ -318,7 +389,312 @@ public static object GetSolarPanelStatus(Vessel vessel)
 
         public static void SolarPanelCtrl(Vessel vessel, bool state)
         {
-            vessel.rootPart.SendEvent(state ? "Extend" : "Retract");
+            foreach (var p in vessel.parts)
+            {
+                foreach (var c in p.FindModulesImplementing<ModuleDeployableSolarPanel>())
+                {
+                    if (state) { c.Extend(); }else { c.Retract(); }
+                }
+            }
+        }
+
+        public static object GetRadiatorStatus(Vessel vessel)
+        {
+            var atLeastOneRadiator = false; // No radiators at all? Always return false
+
+            foreach (var p in vessel.parts)
+            {
+                foreach (var c in p.FindModulesImplementing<ModuleActiveRadiator>())
+                {
+                    atLeastOneRadiator = true;
+
+                    if (! c.IsCooling)
```

Comment:
Why would anybody use toggle on these? Unlike stock action groups which work on toggle, these custom groups can be called (and will cause corresponding action) both ON and OFF no matter the displayed state. If some are activated and some are deactivated, ON will activate what's off and OFF will deactivate what's on